### ISSUE TITLE
Callback must be a function

### ISSUE THREAD

**Issue by ntk860** [sehrope/node-rand-token#7]
Getting this error when generating a random sequence "Callback must be a function"

**Comment by sehrope** [comment]
You're initializing the generator with a `source` property that is not a function. The default is to use `crypto.pseudoRandomBytes(...)` which is almost always the right choice.

If you're still having an issue, post self contained working example that show cases how to reproduce the problem.

**Comment by a7medkamel** [comment]
I ran into this exact same problem 1 day ago.  (my code hadn't changed, just re-installed the module)

narrowed down the problem to this line:

```
return cryptoRandomBytes(size, !cb ? null : function (buf){
            return cb(null, buf);
          });
```

replacing `!cb ? null` with `!cb ? undefined` fixes the problem. I am unclear as to what exactly caused the regression.

It is possible the nodejs has a regression.

calling `c.randomBytes(10, undefined)` works but calling `c.randomBytes(10, null)` throws an exception.

**Comment by igorgo** [comment]
On Node 9.2 executing the `require('rand-token').generate(32)` causes an error:
```
TypeError [ERR_INVALID_CALLBACK]: Callback must be a function
  randomBytes (internal/crypto/random.js:44:11)
  Object.options.source (/rand-token/index.js:61:18)
  Object.generate (/rand-token/index.js:96:29)
```

**Comment by sehrope** [comment]
@igorgo This should be resolved in the latest published version of the module, v0.4.0. If you're still encountering an error then open a new issue and make sure to include your node version and the module version in the description.

**Comment by davidsonsns** [comment]
Really ran @sehrope. I was having the same problem on v9 as the @igorgo said.
